Big tidy up

wub opened the pull request:
Hey guys, fantastic work on the panel!

Here are some changes that should hopefully make it run a bit faster, be a bit more conformant, and with the English slightly improved. This is most probably not a deployable push, as I've modified the reset file (what was up with that! Too aggressive), so when I get home I'll do some testing and do the rest. I have compiled the Javascript - having ~10 requests is just nasty, haha. And it also saves ~200KB. The file is /app/webroot/js/compiled.js.

Also, might want to add no-index/no-follows to the pages, as I have seen a few turn up in Google... I'll add that next push.

<3

wub committed:
Some tidying.

wub commented on app/Controller/InstallController.php:
```diff
@@ -67,17 +69,17 @@ function step1() {
       {
 
-        $sting = '<p class="failed">"app/webroot" folder is not writable</p>';
+        $sting = '<span class="failed">"app/webroot" folder is not writable</span>';
```
Going to further improve these error messages and link them to relevant documentation; hopefully this means people can do it themselves rather than post on the forums about it.

wub commented on app/Locale/deu/LC_MESSAGES/default.po:
```diff
@@ -1313,7 +1313,7 @@ msgid "Server properties"
 
 #: \View\TServersController\index.ctp:31;31
-msgid "Shedules"
+msgid "Schedules"
```
Note that 'Shedules' has been changed to 'Schedules' - if there are any folders by that name, they may be broken... sorry :P

wub commented:
Actually, leave this for now, I will reopen once I've tested it properly.

Antariano commented:
It took me a while to read through all this :P
I have a couple of concerns:

app/Config/database.php
I guess your database settings got pushed, make sure .gitignore is working

Translations are automatically generated by console, so no need to do those :)

Why did you remove the step name from each installation step?

Why did you remove...
-  "by XereoNet | Bukkit Web Administration" ?
- , initial-scale=1.0" from the viewport meta?
-  //GLOBAL    'AdminOnlineCheckTimer' => '',  from configuration.php ?
- app/webroot/app.sql
  the new database tables have space_ as prefix, why did you remove them?

The reset file was not written by us, it's a public domain reset file (don't remember who was the author), and it works, so why change it?

For the rest I welcome these changes :)

wub commented:
Hi there!

Hmm, sorry about that, will recheck the gitignore!

Alright, not too familiar with Cake sorry; what a waste of time! (Translations) Haha.

I removed the names from each installation step as I think they're better suited in a big informative `<h2>` (like I started doing) at the top. It just didn't look right, on OS X, anyway.

I've always felt that titles need not be so long and ungainly - the "by XereoNet" could just be in the footer, and I'm sure they'll know that it's Bukkit Web Administration ;) It's like Facebook saying: "Facebook by Facebook | Social Networking Website."

RE: initial-scale=1.0 https://github.com/h5bp/html5-boilerplate/issues/824

I don't recall removing that //GLOBAL line... perhaps that was a result of my 3AM coding, sorry about that! Can ignore all of the configuration.php changes.

The reason for that database change was because I pulled this before you updated it (I just checked) - didn't know you'd done it sorry!

The reset file was written by Eric Meyer some time ago, and it has been agreed that it is too aggressive - did you notice all of the `<br />`'s that I removed? You had them in there because the reset file was clearing the margins on `<p>` tags! This is not something that should be done - learning to work with the default standards (to a degree) is something that should be done. `<br />` (which, by the way is actually better as `<br>` unless you're using XHTML) tags are to be used sparingly. 

Of course, that's only one example; lists were fully nerfed as well! I am planning a big change (which you can review, of course) that will make the frontend code a lot nicer to work with. Don't pull any changes yet, though! Have you guys used LESSCSS? http://lesscss.org

Also, can you please double check all of that JavaScript? There's quite a bit of it, some paring down might be good, as it's a sizeable ~300kb chunk, ~200kb compressed, and a bit smaller when gzipped.

Thanks Antariano, great panel!

Antariano commented:
Thanks a bunch for clarification :) All this help is highly appreciated!

I never used less, call me old fashioned :P

I didn't know about that reset thing and the viewport, good to know! 

And what is this "big change"? I am intrigued :P

wub commented:
Hey that's no problem at all! The panel has been so much help to me so far.

Haha it's okay; I suggest you look into it though, as it's incredibly powerful and speeds up development time a lot!

The big change will be me stripping a lot of code out, getting LESS set up and really just streamlining the code as much as possible - as well as some aesthetic changes that will make the whole panel feel... snappier! I'll provide more details when I have more code to show :)

Also, LESS can be written in plain CSS, so if anyone doesn't want to use it... they still can! Someone can come through and optimise it later.

Antariano commented:
That's great :)

Good work then!

I suggest you split up your pull request into several small ones, though, to keep things ordered.

wub commented:
Yeah, completely ignore this one, I'll get them sorted properly, sorry about that!
